os/mm/mm_heap/mm_partition_mgr.c

```diff
@@ -0,0 +1,159 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/mm/mm.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+/* If multiple regions are supported or if multi-heap is supported in the project,
+then we dont know how to configure the partitions for user applications. Hence we
+will throw an error here
+*/
+
+#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(__KERNEL__)
+
+#if CONFIG_MM_REGIONS > 1
+#error "Unable to handle APP RAM PARTITIONS for CONFIG_MM_REGIONS > 1"
+#endif
+
+#if CONFIG_MM_NHEAPS > 1
+#error "Unable to handle APP RAM PARTITIONS for CONFIG_MM_NHEAPS > 1"
+#endif
+
+#define ALIGN_EVEN_MASK		2
+#define ALIGN_EVEN(a)		((a) + ALIGN_EVEN_MASK) & ~ALIGN_EVEN_MASK
+/****************************************************************************
+ * Public Variables
+ ****************************************************************************/
+extern uint32_t __ksram_segment_start__[];
+extern uint32_t __ksram_segment_size__[];
+
+/****************************************************************************
+ * Private variables
+ ****************************************************************************/
+static struct mm_partition_s g_mempartitions[CONFIG_NUM_APPS];
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: mm_initialize_partitions
+ *
+ * Description:
+ * This function is called during os boot to initialize the memory partitions.
+ *
+ * Parameters:
+ *   None
+ *
+ * Return Value:
+ *   None
+ *
+ * Assumptions:
+ *
+ ****************************************************************************/
+
+void mm_initialize_partitions(void)
+{
+	int i = 0;
+	uint32_t start;
+	uint32_t size;
+
+	start = (uint32_t)__ksram_segment_start__ + (uint32_t)__ksram_segment_size__;
+	size = (REGION_SIZE - (uint32_t)__ksram_segment_size__) / CONFIG_NUM_APPS;
```

Comment:
All of values are fixed when we compile. Can we make this as definition?

Reply:
I feel it is better to have this as part of the code since it provides type checking.